### ISSUE TITLE
Allow fellow learners to approve solutions

### ISSUE THREAD

**Issue by Jeffrey04** [exercism/discussions#228]
Would it be better if fellow learners get to help approving solutions too instead of relying on a handful of mentors reading through all the submissions? My progress is currently stalled because I am waiting for a mentor's feedback. If reinforcing good programming practice is a concern, requiring more than `N` votes from learners to substitute 1 mentor vote is probably good enough.

**Comment by m3m0ry** [comment]
I agree. Especially with languages which are not highly popular as e.g. python the feedback duration takes too long.

Currently i do Common Lisp. Exersim says there are 0 mentors.

**Comment by kytrinyx** [comment]
Fellow learners are welcome to become mentors: http://mentoring.exercism.io/

**Comment by Jeffrey04** [comment]
lol, but I am not feeling comfortable enough with the language so not going to apply as a mentor hahaha

**Comment by iHiD** [comment]
So the question for me is, if you're not feeling confident enough to
mentor, why would you be confident enough to vote someone's solution as
completed? The two seem to require the same amount of knowledge.

**Comment by Jeffrey04** [comment]
Erm this is exactly why I suggested to require more than 1 (maybe 3 or more?) from non-mentor, if more non-mentor thinks it is fine, then it would most probably be fine. Otherwise the worst case scenario is that everyone will be stuck because there are simply not enough mentors around, and they may all living in different time zones (why are we locking all the exercises again?). Also they are not paid, which means there's nothing we can do to get them on and review every single submissions.

**Comment by blogscot** [comment]
I've just now found myself locked out of the `Rust` track. I imagine at this early stage of Exercism v2 the rate of mentor approvals is going to be far slower than the student submission rate, given there are currently only 9 mentors for 5190 students in this track. By making students to have to wait for approval before moving on, I forsee many finding themselves this same blocked state. 

Would it be an good idea to allow students to progress to the next task after submission (as per Exercism v1), so that the rate of submissions doesn't have to match the rate of approvals?

**Comment by coriolinus** [comment]
That's fundamentally what independent mode is.

**Comment by blogscot** [comment]
Where can I find a description of independent mode?

**Comment by coriolinus** [comment]
I'm not sure if a comprehensive description currently exists. Here's what I know:

- You choose a mode upon joining a track for the first time or upon import from exercism v1
- You can move freely from normal to independent mode, but you cannot move from independent back to normal mode

The button to switch modes is towards the bottom-right of a track's overview page:
![independent mode](https://user-images.githubusercontent.com/7822926/42892641-97f1f504-8ab2-11e8-9d52-e0662b582f42.png)

**Comment by blogscot** [comment]
Thanks @coriolinus, but I'm still not any clearer what the difference between normal and independent mode is, and as there's currently no way back from independent mode I'm not in a hurry to jump in.

**Comment by m3m0ry** [comment]
So i thought about this and also waited a few days.
Having students checking if the solution is not ideal.
I think the main problem is that because of the fresh start of exersism2 there are not yet enough mentors.
The second problem is that most students want to do the exercises one by one, multiple during one session. So they would like to heave immediate feedback, which is impossible, or just take the next exercise.

I think the independent track solves this problem, however it is not stated clearly what the independent mode is! Even i don't know it.

TLDR;
Being in the independent mode, can i request a mentor?
Because this is a crucial feature.

**Comment by m3m0ry** [comment]
This two pieces of information are crucial, please put them somewhere on the site for everyone to see.

**Comment by iHiD** [comment]
As I said, we're working on making both of these clearer as a priority.
Both pieces of information are on the site but people don't always read
everything, so we have to find more inventive ways to make these things
clearer. We're working on it :)

**Comment by rpottsoh** [comment]
In independent mode you do not have to wait for mentor approval before moving on the next core exercise.  I think also (and this is where I am a little fuzzy on the topic) I believe those students that have chosen the independent route, their solutions are not public.  I also think they may not be able to see submissions from other students as well, since there submissions are not public.  I am not sure if, as an independent student, you are able to selectively publicize your submissions which would then in turn grant you access to the public submissions and vice versa….

**Comment by pranasas** [comment]
This is clearly still an issue. I wish to enjoy the default experience but the main Python Track is locked (it's quite a popular one, isn't it?) and the last community solution was published 11 days ago.

**Comment by iHiD** [comment]
> This is clearly still an issue. I wish to enjoy the default experience but the main Python Track is locked (it's quite a popular one, isn't it?) and the last community solution was published 11 days ago.

Right now there are 15 unmentored exercises on Python. Multiple core exercises have been approved today. So I don't think it is an issue. Presuming you've completed Hello World (which doesn't require mentor approval) there are about 20 exercises you can do under the Extra Exercises section while you wait for mentoring on your core exercise.

I'm going to close this issue as this isn't a feature we're going to introduce. Thanks for suggesting it though :)
